README.rst
QRainbowStyleSheet
==================

|Latest PyPI version| |Supported python versions| |Build Status| |Docs Status| |License: MIT|
|License: CC BY 4.0| |Conduct|

The most complete customizable stylesheet for Qt application (PySide2,
PyQt5, QtPy, Qt.Py).

Preview
-------

From version v0.8 qrainbowstyle.windows module supports native Windows calls.
Features:

    - Borders snapping
    - Minimize, restore, close animations
    - Size grips on borders
    - Frame shadow
    - Aero shake

On Linux and Darwin qrainbowstyle will load class with its own implementation of these features.
Due to a bug in Qt, window minimizing is not supported on MacOS.


Frameless windows
~~~~~~~~~~~~~~~~~

.. image:: https://raw.githubusercontent.com/desty2k/QRainbowStyleSheet/master/images/frameless_window_v3.png


Installation
------------

From PyPI: Get the latest stable version of ``qrainbowstyle`` package using
*pip* (preferable):

    .. code:: bash

        pip install qrainbowstyle


From code: Download/clone the project, go to ``qrainbowstyle`` folder then:

-  You can use the *setup* script and pip install.

    .. code:: bash

        pip install .


-  Or, you can use the *setup* script with Python:

    .. code:: bash

        python setup.py install


Usage
-----


Frameless windows
~~~~~~~~~~~~~~~~~~
.. code:: python

    import os
    import sys
    import qrainbowstyle
    import qrainbowstyle.windows

    from qtpy.QtWidgets import QApplication
    from qtpy.QtCore import Qt

    QApplication.setAttribute(Qt.AA_UseHighDpiPixmaps)
    QApplication.setAttribute(Qt.AA_EnableHighDpiScaling)

    app = QApplication(sys.argv)
    app.setStyleSheet(qrainbowstyle.load_stylesheet(style="oceanic"))

    # Package options
    # qrainbowstyle.alignButtonsLeft()
    # qrainbowstyle.userDarwinButtons()
    qrainbowstyle.setAppName("My new application")
    qrainbowstyle.setAppIcon("/path/to/icon.ico")

    # Create frameless mainwindow
    win = qrainbowstyle.windows.FramelessWindow()

    # Example of using signals
    win.closeClicked.connect(lambda: print("Close clicked!"))

    # Create content widget and pass reference to main window
    widget = SomeWidget(win)

    # Add widget to main window and show it
    win.addContentWidget(widget)
    win.show()

    sys.exit(app.exec())


Style sheet
~~~~~~~~~~~~
.. code:: python

    import os
    import sys
    import qrainbowstyle

    # set the environment variable to use a specific wrapper
    # it can be set to pyqt, pyqt5, or pyside2
    # you do not need to use QtPy to set this variable
    os.environ['QT_API'] = 'pyqt5'

    # import from QtPy instead of doing it directly
    # note that QtPy always uses PyQt5 API
    from qtpy import QtWidgets

    # create the application and the main window
    app = QtWidgets.QApplication(sys.argv)
    window = QtWidgets.QMainWindow()

    # setup stylesheet
    # the default system in qrainbowstyle uses qtpy environment variable
    app.setStyleSheet(qrainbowstyle.load_stylesheet())

    # run
    window.show()
    app.exec_()


If you are using PyQt5 directly, see the complete example

.. code:: python

    import sys
    import qrainbowstyle
    from PyQt5.QtWidgets import QApplication, QMainWindow

    # create the application and the main window
    app = QApplication(sys.argv)
    window = QMainWindow()

    # setup stylesheet
    app.setStyleSheet(qrainbowstyle.load_stylesheet_pyqt5())
    # or in new API
    app.setStyleSheet(qrainbowstyle.load_stylesheet(qt_api='pyqt5'))

    # run
    window.show()
    app.exec_()


Here is an example using PySide2

.. code:: python

    import sys
    import qrainbowstyle
    from Pyside2.QtWidgets import QApplication, QMainWindow

    # create the application and the main window
    app = QApplication(sys.argv)
    window = QMainWindow()

    # setup stylesheet
    app.setStyleSheet(qrainbowstyle.load_stylesheet_pyside2())
    # or in new API
    app.setStyleSheet(qrainbowstyle.load_stylesheet(qt_api='pyside2'))

    # run
    window.show()
    app.exec_()


If you are using Qt.py, which is different from qtpy, you should install
qtpy then set both to the same binding.


Available styles
----------------

Currently available styles are:

* Oceanic
* QDarkStyle3
* QDarkstyle3Light
* LightOrange
* DarkOrange
* QDarkStyle (original)

Select style by using ``style=`` keyword argument

.. code:: python

    qrainbowstyle.load_stylesheet(style="lightorange")


Widgets
-------

In v0.6 I added a few new widgets which automatically load colors from current
stylesheet's palette.


GoogleMapsView
~~~~~~~~~~~~~~

GoogleMapsView allows to load Google Maps to application. Supports creating markers and polylines.
Call handler captures all map/markers/polylines actions such as map move or marker click/double click.

.. image:: https://raw.githubusercontent.com/desty2k/QRainbowStyleSheet/master/images/frameless_mainwindow_google_maps_example.png


StylePicker
~~~~~~~~~~~

StylePicker is small widget used to change stylesheet without restarting application.
Only styles generated by QRainbowStyle are supported. Widget is available in 3 versions: horizontal, vertical and grid.

.. image:: https://raw.githubusercontent.com/desty2k/QRainbowStyleSheet/master/images/frameless_mainwindow_color_picker_example.png


QRoundProgressBar
~~~~~~~~~~~~~~~~~

Modified version of https://github.com/ozmartian/QRoundProgressBar.
I replaced PyQt5 imports with qtpy and fixed widget background.

.. image:: https://raw.githubusercontent.com/desty2k/QRainbowStyleSheet/master/images/frameless_mainwindow_round_progress_bar.png


QtWaitingSpinner
~~~~~~~~~~~~~~~~~

Modified version of https://github.com/fbjorn/QtWaitingSpinner. Added fade out and fade in.
Spinner designer can be found in `qrainbowstyle/widgets/QtWaitingSpinner/designer.py`

.. image:: https://raw.githubusercontent.com/desty2k/QRainbowStyleSheet/master/images/waiting_spinner_designer.png


Building your own style sheet
-----------------------------

Download/clone the project, go to ``qrainbowstyle`` folder then:

1. Create new style in palette.py by subclassing BasePalette. New palette should have unique name, for example ``DeepBluePalette``

2. Override default colors by your own. Example:

    .. code:: python

        class DeepBluePalette(BasePalette):
            COLOR_BACKGROUND_1 = "#263238"
            COLOR_BACKGROUND_2 = "#2f4048"
            COLOR_BACKGROUND_3 = "#34474f"
            COLOR_BACKGROUND_4 = "#394d57"
            COLOR_BACKGROUND_5 = "#3d545f"
            COLOR_BACKGROUND_6 = "#425b67"

            COLOR_TEXT_1 = Gray.B130
            COLOR_TEXT_2 = Gray.B110
            COLOR_TEXT_3 = Gray.B90
            COLOR_TEXT_4 = Gray.B80

            COLOR_ACCENT_1 = "#0a4542"
            COLOR_ACCENT_2 = "#136460"
            COLOR_ACCENT_3 = "#097D74"
            COLOR_ACCENT_4 = "#56BFBA"
            COLOR_ACCENT_5 = "#C4D6DB"

            TITLE_BAR_BACKGROUND_COLOR = COLOR_ACCENT_3
            TITLE_BAR_BUTTONS_HOVER_COLOR = COLOR_ACCENT_4
            TITLE_BAR_BUTTONS_DISABLED_COLOR = COLOR_ACCENT_1
            TITLE_BAR_TEXT_COLOR = COLOR_TEXT_1

            OPACITY_TOOLTIP = 230

3. Generate resources for your style by running scripts/process_qrc.py

4. Install package by running:

    .. code:: python

        pip install .

5. To use style sheet in your application:

    .. code:: python

        import qrainbowstyle

        app = QApplication(sys.argv)
        app.setStyleSheet(qrainbowstyle.load_stylesheet(style="deepblue")


What is new?
------------

Starting with new package name, I added frameless widnows and possibility to
design your own stylesheet. I added few new SVG icons such as title bar icons.

- 0.6 - widget subpackage
- 0.8 - Windows API support
- 0.9 - modern style for frameless windows, full support for new color system


Changelog
---------

Please, see `CHANGES <CHANGES.rst>`__ file.


License
-------

This project is licensed under the MIT license. Images contained in this
project are licensed under CC-BY license.

For more information see `LICENSE <LICENSE.rst>`__ file.


Authors
-------

For more information see `AUTHORS <AUTHORS.rst>`__ file.


Contributing
------------

Most widgets have been styled. If you find a widget that has not been
style, just open an issue on the issue tracker or, better, submit a pull
request.

If you want to contribute, see `CONTRIBUTING <CONTRIBUTING.rst>`__ file.

.. |Build Status| image:: https://github.com/desty2k/QRainbowStyleSheet/workflows/build/badge.svg
   :target: https://github.com/desty2k/QRainbowStyleSheet/actions?workflow=build
.. |Docs Status| image:: https://github.com/desty2k/QRainbowStyleSheet/workflows/docs/badge.svg
   :target: https://desty2k.github.io/QRainbowStyleSheet/
.. |Latest PyPI version| image:: https://img.shields.io/pypi/v/QRainbowStyle.svg
   :target: https://pypi.org/project/QRainbowStyle/
.. |Supported python versions| image:: https://img.shields.io/pypi/pyversions/QRainbowStyle.svg
   :target: https://pypi.org/project/QRainbowStyle/
.. |License: MIT| image:: https://img.shields.io/dub/l/vibe-d.svg?color=lightgrey
   :target: https://opensource.org/licenses/MIT
.. |License: CC BY 4.0| image:: https://img.shields.io/badge/License-CC%20BY%204.0-lightgrey.svg
   :target: https://creativecommons.org/licenses/by/4.0/
.. |Conduct| image:: https://img.shields.io/badge/code%20of%20conduct-contributor%20covenant-green.svg?style=flat&color=lightgrey
   :target: https://www.contributor-covenant.org/version/2/0/code_of_conduct/


CHANGES.rst
Changelog
=========

- 0.9.6:
    - Cast ``floats`` to ``ints`` to fix TypeErrors in new Qt version
    - Install Qt message handler in spinner designer
    - Fix typos

- 0.9.5:
    - Fix style selector alignment in spinner designer
    - Remove ``Create a first release`` step from build workflow
    - Reformat GitHub Actions workflows
    - Remove pip caching
    - Remove unused ``develop`` tox action
    - Update setup-python action to 2.3.2
    - Update changelog-generator action to 3.0.0
    - Update info about ``load_stylesheet()`` in README

- 0.9.4:
    - Generate changelog when creating new GitHub release

- 0.9.3:
    - Generate docs using apidoc
    - Remove unused files from widgets package
    - Use moduels instead of packages where possible

- 0.9.2:
    - Update docs
    - Update issue template
    - Update contributing guide
    - Clean the code
    - Docs workflow will be run only after creating release
    - Update release action

- 0.9.1:
    - Fix child windows being minimized to left bottom corner
    - Set titlebar buttons' icons using stylesheet
    - Add icons for titlebar buttons disabled state
    - Add setWindowFlags and setContentMargins methods
    - Generate resources for styles
    - Update readme

- 0.9:
    - Redesigned frameless windows
    - Add color system from qdarkstyle v3
    - Add new preview images
    - Clean unused files and code
    - Generate resources

- 0.8.3:
    - Fix setting parent for frameless windows, window modality works now
    - Fix window outside display when moved in maximized mode
    - Fix do not list pycache as style
    - Update example

- 0.8.2:
    - Do not use app event filters in FramelessWindows
      for non Windows OS, fixes 0xC0000005 errors
    - Update PyPI docs and readme
    - Fix moving window not working in frameless messageboxes

- 0.8.1:
    - Fix file with capitalized name not commited to GH
    - Do not use titlebar signal to close window

- 0.8:
    - Refactor code in windows module
    - Add Windows API support
    - Add OS detection
    - Update funcs names
    - Update examples
    - Update scripts to use loggers
    - 0xC0000005 errors are now fixed
    - Regenerate resources

- 0.7.1:
    - Fix windows being deleted after creating them in slots
    - Reduce amount of window flags

- 0.7:
    - Add color pallette from qdarkstyle v3
    - Update checkbox icon when checked
    - Update styles using new palette
    - Generate resources
    - Update readme
    - Update preview images
    - Update examples
    - Add slot to connect two markers in GoogleMapsView
    - Fix marker removed after double clicking
    - Remove prints on fade in/out in spinner

- 0.6.6:
    - Add setters for window title and icon
    - Update readme
    - Update OutputLogger
    - Update example2
    - Update loggers names
    - Fix white background when resizing GoogleMapsView

- 0.6.5:
    - Add button getter is FramelessMessageBox
    - GoogleMapsView: Add Python boolean to JS converter
    - GoogleMapsView: Add map events catchers - resizing/loading
    - GoogleMapsView: Add 4 new map settings
    - GoogleMapsView: Add map function: pan to center
    - Update package logger name
    - Fix chdir back to working dir after importing style
    - Update GoogleMapsView example

- 0.6.4:
    - Add vertical and horizontal style pickers
    - Add fadeIn and fadeOut for waitingspinner
    - Change StylePicker name to StylePickerGrid
    - Update frameless mainwindow gif
    - Fix html not loaded when running javascript scripts

- 0.6.3:
    - Add preview gifs
    - Add docs for titlebar
    - Add space between window side and buttons in titlebar
    - Remove getWorkspace titlebar
    - Fix buttons appearing in titlebar even when using darwin style

- 0.6.2:
    - Fix resize handler
    - Add FramelessDialog and FramelessMessageBoxes
    - GoogleMapsView: support for markers and polylines
    - GoogleMapsView: logging Javascript console errors
    - GoogleMapsView: signals for mouse events
    - GoogleMapsView: update example3
    - Add FramelessDialog to style.scss
    - Re-generate resources
    - Replace .pngs with .gif in README in preview section

- 0.6.1:
    - Add QRoundProgressBar
    - Add example for round progress bar
    - Fix error when importing widgets
    - Update README

- 0.6:
    - Add early versions of GoogleMapsView and StylePicker
    - Add docs for new widgets
    - Add example scripts for new widgets
    - Add new preview images
    - Update README

- 0.5.6:
    - Add Resizer module
    - Add rainbowize() and get_available_palettes()
    - Add getters for screen geometry
    - Add setIcons() for titlebar icons
    - Fix updating titlebar buttons' icons after changing stylesheet
    - Move debugging stuff to extras
    - Remove content widget margins for frameless window

- 0.5.5:
    - Add widgets subpackage
    - Add QtWaitingSpinner
    - Update spinner to use qtpy
    - Update spinner to use color from palette
    - Update spinner designer
    - Remove deploy on push with v* tag
    - Fix window flickering when resizing frameless windows
    - Add custom frameless windows resize
    - Remove size grip
    - Update example2

- 0.5.4:
    - Fix copyrights in license
    - Add square icons for close buttons
    - Generate resources

- 0.5.3:
    - Add auto publishing release on successful build
    - Update docs url in setup.py

- 0.5.2:
    - Add auto release to PyPI after build
    - Ignore styles directory in Pylint

-  0.5.1:
    - Update PyPI docs
    - Update badges
    - Fix deploy workflow

-  0.5:
    -  Add build and docs workflows
    -  Add badges to README
    -  Add PyPI deploy workflow
    -  Create first QRainbowStyleSheet release
    -  Upload package to PyPI
    -  Update comments

qrainbowstyle/palette.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""QRainbowStyle default palette."""

# Standard library imports
from collections import OrderedDict

# package imports
from qrainbowstyle.colorsystem import *


class BasePalette:
    """Base class for palettes."""

    # Color
    COLOR_BACKGROUND_1 = ''
    COLOR_BACKGROUND_2 = ''
    COLOR_BACKGROUND_3 = ''
    COLOR_BACKGROUND_4 = ''
    COLOR_BACKGROUND_5 = ''
    COLOR_BACKGROUND_6 = ''

    COLOR_TEXT_1 = ''
    COLOR_TEXT_2 = ''
    COLOR_TEXT_3 = ''
    COLOR_TEXT_4 = ''

    COLOR_ACCENT_1 = ''
    COLOR_ACCENT_2 = ''
    COLOR_ACCENT_3 = ''
    COLOR_ACCENT_4 = ''
    COLOR_ACCENT_5 = ''

    OPACITY_TOOLTIP = 0

    # Size
    SIZE_BORDER_RADIUS = '4px'

    # Borders
    BORDER_1 = '1px solid $COLOR_BACKGROUND_1'
    BORDER_2 = '1px solid $COLOR_BACKGROUND_4'
    BORDER_3 = '1px solid $COLOR_BACKGROUND_6'

    BORDER_SELECTION_3 = '1px solid $COLOR_ACCENT_3'
    BORDER_SELECTION_2 = '1px solid $COLOR_ACCENT_2'
    BORDER_SELECTION_1 = '1px solid $COLOR_ACCENT_1'

    TITLE_BAR_BACKGROUND_COLOR = COLOR_ACCENT_3
    TITLE_BAR_BUTTONS_HOVER_COLOR = COLOR_ACCENT_4
    TITLE_BAR_BUTTONS_DISABLED_COLOR = COLOR_ACCENT_1
    TITLE_BAR_TEXT_COLOR = COLOR_TEXT_1

    # Paths
    PATH_RESOURCES = "':/qss_icons'"

    @classmethod
    def to_dict(cls, colors_only=False):
        """Convert variables to dictionary."""
        order = [
            'COLOR_BACKGROUND_6',
            'COLOR_BACKGROUND_5',
            'COLOR_BACKGROUND_4',
            'COLOR_BACKGROUND_2',
            'COLOR_BACKGROUND_3',
            'COLOR_BACKGROUND_1',
            'COLOR_TEXT_1',
            'COLOR_TEXT_2',
            'COLOR_TEXT_3',
            'COLOR_TEXT_4',
            'COLOR_ACCENT_1',
            'COLOR_ACCENT_2',
            'COLOR_ACCENT_3',
            'COLOR_ACCENT_4',
            'OPACITY_TOOLTIP',
            'SIZE_BORDER_RADIUS',
            'BORDER_1',
            'BORDER_2',
            'BORDER_3',
            'BORDER_SELECTION_3',
            'BORDER_SELECTION_2',
            'BORDER_SELECTION_1',
            'TITLE_BAR_BACKGROUND_COLOR',
            'TITLE_BAR_BUTTONS_HOVER_COLOR',
            'TITLE_BAR_BUTTONS_DISABLED_COLOR',
            'TITLE_BAR_TEXT_COLOR',
            'PATH_RESOURCES',
        ]
        dic = OrderedDict()
        for var in order:
            value = getattr(cls, var)

            if colors_only:
                if not var.startswith('COLOR'):
                    value = None

            if value:
                dic[var] = value

        return dic

    @classmethod
    def color_palette(cls):
        """Return the ordered colored palette dictionary."""
        return cls.to_dict(colors_only=True)


class Oceanic(BasePalette):
    """Theme variables."""
    COLOR_BACKGROUND_1 = "#263238"
    COLOR_BACKGROUND_2 = "#2f4048"
    COLOR_BACKGROUND_3 = "#34474f"
    COLOR_BACKGROUND_4 = "#394d57"
    COLOR_BACKGROUND_5 = "#3d545f"
    COLOR_BACKGROUND_6 = "#425b67"

    COLOR_TEXT_1 = Gray.B130
    COLOR_TEXT_2 = Gray.B110
    COLOR_TEXT_3 = Gray.B90
    COLOR_TEXT_4 = Gray.B80

    COLOR_ACCENT_1 = "#0a4542"
    COLOR_ACCENT_2 = "#136460"
    COLOR_ACCENT_3 = "#097D74"
    COLOR_ACCENT_4 = "#56BFBA"
    COLOR_ACCENT_5 = "#C4D6DB"

    TITLE_BAR_BACKGROUND_COLOR = COLOR_ACCENT_3
    TITLE_BAR_BUTTONS_HOVER_COLOR = COLOR_ACCENT_4
    TITLE_BAR_BUTTONS_DISABLED_COLOR = COLOR_ACCENT_1
    TITLE_BAR_TEXT_COLOR = COLOR_TEXT_1

    OPACITY_TOOLTIP = 230


class QDarkStyle(BasePalette):
    COLOR_BACKGROUND_1 = "#19232d"
    COLOR_BACKGROUND_2 = "#27323c"
    COLOR_BACKGROUND_3 = "#35414b"
    COLOR_BACKGROUND_4 = "#3e4b55"
    COLOR_BACKGROUND_5 = "#47555f"
    COLOR_BACKGROUND_6 = "#505f69"

    COLOR_TEXT_1 = Gray.B130
    COLOR_TEXT_2 = Gray.B110
    COLOR_TEXT_3 = Gray.B90
    COLOR_TEXT_4 = Gray.B80

    COLOR_ACCENT_1 = "#14506e"
    COLOR_ACCENT_2 = "#145f87"
    COLOR_ACCENT_3 = "#146998"
    COLOR_ACCENT_4 = "#1478b1"
    COLOR_ACCENT_5 = "#148cd2"

    TITLE_BAR_BACKGROUND_COLOR = COLOR_ACCENT_3
    TITLE_BAR_BUTTONS_HOVER_COLOR = COLOR_ACCENT_4
    TITLE_BAR_BUTTONS_DISABLED_COLOR = COLOR_ACCENT_1
    TITLE_BAR_TEXT_COLOR = COLOR_TEXT_1

    OPACITY_TOOLTIP = 230


class DarkOrange(BasePalette):
    COLOR_BACKGROUND_1 = Gray.B10
    COLOR_BACKGROUND_2 = Gray.B20
    COLOR_BACKGROUND_3 = Gray.B30
    COLOR_BACKGROUND_4 = Gray.B40
    COLOR_BACKGROUND_5 = Gray.B50
    COLOR_BACKGROUND_6 = Gray.B60

    COLOR_TEXT_1 = Gray.B130
    COLOR_TEXT_2 = Gray.B110
    COLOR_TEXT_3 = Gray.B90
    COLOR_TEXT_4 = Gray.B80

    COLOR_ACCENT_1 = "#ce4b01"
    COLOR_ACCENT_2 = "#d66522"
    COLOR_ACCENT_3 = "#de8044"
    COLOR_ACCENT_4 = "#e79b65"
    COLOR_ACCENT_5 = "#efb587"

    TITLE_BAR_BACKGROUND_COLOR = COLOR_ACCENT_3
    TITLE_BAR_BUTTONS_HOVER_COLOR = COLOR_ACCENT_4
    TITLE_BAR_BUTTONS_DISABLED_COLOR = COLOR_ACCENT_1
    TITLE_BAR_TEXT_COLOR = COLOR_TEXT_1

    OPACITY_TOOLTIP = 230


class LightOrange(BasePalette):
    COLOR_BACKGROUND_1 = Gray.B140
    COLOR_BACKGROUND_2 = Gray.B130
    COLOR_BACKGROUND_3 = Gray.B120
    COLOR_BACKGROUND_4 = Gray.B110
    COLOR_BACKGROUND_5 = Gray.B100
    COLOR_BACKGROUND_6 = Gray.B90

    COLOR_TEXT_1 = Gray.B10
    COLOR_TEXT_2 = Gray.B20
    COLOR_TEXT_3 = Gray.B50
    COLOR_TEXT_4 = Gray.B70

    COLOR_ACCENT_1 = "#ce4b01"
    COLOR_ACCENT_2 = "#d66522"
    COLOR_ACCENT_3 = "#de8044"
    COLOR_ACCENT_4 = "#e79b65"
    COLOR_ACCENT_5 = "#efb587"

    TITLE_BAR_BACKGROUND_COLOR = COLOR_ACCENT_3
    TITLE_BAR_BUTTONS_HOVER_COLOR = COLOR_ACCENT_4
    TITLE_BAR_BUTTONS_DISABLED_COLOR = COLOR_ACCENT_1
    TITLE_BAR_TEXT_COLOR = COLOR_TEXT_1

    OPACITY_TOOLTIP = 230


class QDarkStyle3Light(BasePalette):
    # Color
    COLOR_BACKGROUND_1 = Gray.B140
    COLOR_BACKGROUND_2 = Gray.B130
    COLOR_BACKGROUND_3 = Gray.B120
    COLOR_BACKGROUND_4 = Gray.B110
    COLOR_BACKGROUND_5 = Gray.B100
    COLOR_BACKGROUND_6 = Gray.B90

    COLOR_TEXT_1 = Gray.B10
    COLOR_TEXT_2 = Gray.B20
    COLOR_TEXT_3 = Gray.B50
    COLOR_TEXT_4 = Gray.B70

    COLOR_ACCENT_1 = Blue.B130
    COLOR_ACCENT_2 = Blue.B100
    COLOR_ACCENT_3 = Blue.B90
    COLOR_ACCENT_4 = Blue.B80
    COLOR_ACCENT_5 = Blue.B70

    TITLE_BAR_BACKGROUND_COLOR = COLOR_ACCENT_3
    TITLE_BAR_BUTTONS_HOVER_COLOR = COLOR_ACCENT_4
    TITLE_BAR_BUTTONS_DISABLED_COLOR = COLOR_ACCENT_1
    TITLE_BAR_TEXT_COLOR = COLOR_TEXT_1

    OPACITY_TOOLTIP = 230


class QDarkStyle3(BasePalette):
    # Color
    COLOR_BACKGROUND_1 = Gray.B10
    COLOR_BACKGROUND_2 = Gray.B20
    COLOR_BACKGROUND_3 = Gray.B30
    COLOR_BACKGROUND_4 = Gray.B40
    COLOR_BACKGROUND_5 = Gray.B50
    COLOR_BACKGROUND_6 = Gray.B60

    COLOR_TEXT_1 = Gray.B130
    COLOR_TEXT_2 = Gray.B110
    COLOR_TEXT_3 = Gray.B90
    COLOR_TEXT_4 = Gray.B80

    COLOR_ACCENT_1 = Blue.B20
    COLOR_ACCENT_2 = Blue.B40
    COLOR_ACCENT_3 = Blue.B50
    COLOR_ACCENT_4 = Blue.B70
    COLOR_ACCENT_5 = Blue.B80

    TITLE_BAR_BACKGROUND_COLOR = COLOR_ACCENT_3
    TITLE_BAR_BUTTONS_HOVER_COLOR = COLOR_ACCENT_4
    TITLE_BAR_BUTTONS_DISABLED_COLOR = COLOR_ACCENT_1
    TITLE_BAR_TEXT_COLOR = COLOR_TEXT_1

    OPACITY_TOOLTIP = 230
